src/components/Form.jsx

```diff
@@ -1,96 +1,78 @@
-
-  const [isSubmitBtnDisabled, setIsSubmitBtnDisabled] = React.useState(true);
-
-  React.useEffect(() => {
-    const isValid = isFormValid(formData);
-    setIsSubmitBtnDisabled(!isValid);
-  }, [formData]);
-
-  const handleChange = (event) => {
-    const { name, type, checked } = event.target;
-    let { value } = event.target;
-    value = type === 'checkbox' ? checked : value;
-    setFormData((data) => ({ ...data, [name]: { ...data[name], value } }));
-  };
-
-  const validate = (name, type, value) => {
-    const error = validateForm(value, type);
-    setFormData((data) => ({ ...data, [name]: { ...data[name], error } }));
-    if (type === 'tel' && !error) {
-      const tel = telFormatted(value);
-      setFormData((data) => ({ ...data, [name]: { value: tel, error: '' } }));
-    }
-  };
-
-  const handleSubmit = (event) => {
+const Form = ({
+  searchValue,
+  sortBy,
+  pageSize,
+  currentPage,
+  isLoading,
+  setCards,
+  setIsLoading,
+  setSearchValue,
+  setSortBy,
+  setPageSize,
+  setTotalPages,
+  setCurrentPage,
+  setIsPaginatorBtnClicked,
+}) => {
+  const handleSubmit = async (event) => {
     event.preventDefault();
-    const newCard = {
-      name: `${formData.firstName.value} ${formData.lastName.value}`,
-      phone: formData.phone.value,
-      deliveryDate: formData.deliveryDate.value,
-      country: formData.country.value,
-      gender: formData.gender.value,
-    };
-    setCards((cards) => [...cards, newCard]);
-    setFormData(initFormData);
-    // eslint-disable-next-line no-alert
-    alert('Data sent successfully!');
+    setIsPaginatorBtnClicked(false);
+    setIsLoading(true);
+    try {
+      const response = await axiosInstance.get('v2/everything', {
+        params: {
+          q: searchValue,
+          sortBy,
+          pageSize,
+          page: defaultCurrentPage,
+        },
+      });
+      const { articles } = response.data;
+      if (articles.length === 0) {
+        setCards(`Sorry, your search "${searchValue}" did not return any results.`);
+      } else {
+        const formattedResponse = articles.map(
+          ({ author, title, description, publishedAt, source, url, urlToImage }) => ({
+            author,
+            title,
+            description: formattedDescription(description),
+            publishedAt: formattedDate(publishedAt),
+            source: source.name,
+            url,
+            urlToImage,
+          })
+        );
+        setCards(formattedResponse);
+        const totalPages = Math.ceil(response.data.totalResults / pageSize);
+        if (currentPage !== defaultCurrentPage) {
+          setCurrentPage(defaultCurrentPage);
+        }
+        setTotalPages(totalPages);
+      }
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error(error);
+      // eslint-disable-next-line no-alert
+      alert(error);
+    } finally {
+      setIsLoading(false);
+    }
```

Comment:
a lot of logic for submitting handler: axios, nesting if statements, response treatment